src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -340,10 +340,11 @@ private BoundExpression ConvertSwitchExpression(BoundUnconvertedSwitchExpression
             {
                 var oldCase = source.SwitchArms[i];
                 var oldValue = oldCase.Value;
+                var binder = GetRequiredBinder(oldCase.Syntax);
                 var newValue =
                     targetTyped
-                    ? CreateConversion(oldValue.Syntax, oldValue, underlyingConversions[i], isCast: false, conversionGroupOpt: null, destination, diagnostics)
-                    : GenerateConversionForAssignment(destination, oldValue, diagnostics);
+                    ? binder.CreateConversion(oldValue.Syntax, oldValue, underlyingConversions[i], isCast: false, conversionGroupOpt: null, destination, diagnostics)
```

Comment:
I couldn't follow how using a different binder for conversion could result in the original `The name 'n' does not exist in the current context` error (which doesn't seem related to conversion). How does this work?

Reply:
The switch expression is the only expression that introduces locals in its own scope, so we need the actual binder of the switch arms here to get access to those scoped locals while binding `new()` initializer which happens during the conversion.